packages/itwinui-react/src/core/Popover/Popover.tsx

```diff
@@ -275,13 +296,70 @@ export const usePopover = (options: PopoverOptions & PopoverInternalProps) => {
       ...interactions,
       getFloatingProps,
       ...floating,
+      interactionsEnabledStates,
     }),
-    [open, onOpenChange, interactions, getFloatingProps, floating],
+    [
+      open,
+      onOpenChange,
+      interactions,
+      getFloatingProps,
+      floating,
+      interactionsEnabledStates,
+    ],
   );
 };
 
 // ----------------------------------------------------------------------------
 
+/**
+ * @private
+ * Helper hook to generate the necessary props for
+ * Floating UI's [listNavigation](https://floating-ui.com/docs/useListNavigation) interaction in `usePopover`.
+ *
+ * This helper hook automatically takes care of providing `listRef` and `activeIndex`.
+ * It also updates the `activeIndex` when the user navigates through the list using the `listNavigation` interaction.
+ *
+ * @param props Will be merged with the default props.
+ * @returns The props to spread on the `usePopover.interactions.listNavigation`.
+ *
+ * @example
+ * const listNavigationProps = useListNavigationProps({ nested: subMenuItems.length > 0 });
+ * const popover = usePopover({ interactions: { listNavigation: listNavigationProps })
+ */
+export const useListNavigationProps = (
+  props?: Partial<Omit<UseListNavigationProps, 'activeIndex' | 'listRef'>>,
+): UseListNavigationProps => {
+  const [currentFocusedNodeIndex, setCurrentFocusedNodeIndex] = React.useState<
+    number | null
+  >(null);
+  const focusableNodes = React.useRef<Array<HTMLElement | null>>([]);
+
+  const {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    activeIndex: activeIndexProp,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    listRef: listRefProp,
```

Comment:
I'm not sure I understand this. You're accepting these props only to then ignore them?

Reply:
I was hoping to make sure that `useListNavigationProps`'s `activeIndex` and `listRef` are not overridden by what is passed in its props. I think this was important when I was experimenting with a `useEffect` inside `useListNavigationProps` that was editing the `listRef` internally. But since I don't have that `useEffect` anymore, I don't think this is too important.

So, upon re-reading that code, I believe this JS safeguard is unnecessary and that the TS types are sufficient to prevent overriding those two props. Thus, removed that JS safeguard (https://github.com/iTwin/iTwinUI/pull/2021/commits/8fd3dabe68212da396b6a71dfb6d890d458872e5).